### ISSUE TITLE
AuthProvider service type

### ISSUE THREAD

**Issue by Cafeinoman** [vert-x3/vertx-service-discovery#26]
Is it planned to add a type to be able to discover AuthProvider service. It will be useful for a centralized auth in microservice architecture...
If you are interested, but not have time, I can work on this...

**Comment by cescoffier** [comment]
👍 

I like the idea. It was not planned, but definitely something we can add. Notice that _service types_ are pluggable, so we can implement it in a separate project (or module under this project).

**Comment by cescoffier** [comment]
@Cafeinoman just a question, what do you call `AuthProvider`, is it https://github.com/vert-x3/vertx-auth/blob/master/vertx-auth-common/src/main/java/io/vertx/ext/auth/AuthProvider.java ?

**Comment by Cafeinoman** [comment]
Yes, that's it. I begin to work on a (very) simple implementation couple weeks ago, but got a lot work. Missing a lot of test at this time. You can find it on my repo.

**Comment by cescoffier** [comment]
Just add a look. It looks good to me. Yes need more tests (especially polyglot test to be sure it's usable from groovy, js and ruby).
